### PR TITLE
Boundary from file and new footprints

### PR DESCRIPTION
**Pull request by wehrad**
Implementation proposal of a new condition to set footprint from a file containing desired coordinates (.csv and .geojson implemented), using boundary_from_file.py. Besides, footprints for different study areas have been hard-coded.

### CODE REVIEW

**Comment by mankoff** [comment]
Hi Adriene - looks good but please make the following changes:

Add a note, comment, or documentation about the format of the CSV file. Is it `lon, lat` or `lat, lon`? And I assume decimal degrees? I don't know GeoJSON but if that is a loosely-defined format and I need to know that in order to use your code, let me know that too.

I think you can also squash multiple commits to one, so that the revisions history stays cleaner, but don't worry about this now.

File: boundary_from_file.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    
+    #extension extraction out of file name
+    extension=file.split('.')[-1]
+    
+    #data loading
+    if extension=='geojson':
+        with open(file,encoding='utf8') as f:
+            data = json.load(f)
+            coordinates=data['features'][0]['geometry']['coordinates']
+    if extension=='csv':
```
Comment (wehrad) [inline boundary_from_file.py:26]: elif extension=='csv':

File: dhusget_wrapper.sh [hunk 2]
```diff
@@ -16,7 +16,7 @@ print_usage() {
        "[dhusget.sh options]"
   echo ""
   echo "  -d|--date YYYY-MM-DD or YYY-DOY"
-  echo "  [-f|--footprint Greenland|Iceland|<footprint code> [DEFAULT: Greenland]]"
+  echo "  [-f|--footprint Greenland|Iceland|Svalbard|NovayaZemlya|FransJosefLand|ArcticCanada [DEFAULT: Greenland] [ELSE: footprint coordinates (.geojson,.csv file)]]"
```
Comment (wehrad) [inline dhusget_wrapper.sh]: echo "  [-f|--footprint Greenland|Iceland|Svalbard|NovayaZemlya|FransJosefLand|ArcticCanada [DEFAULT: Greenland] [ELSE: footprint coordinates (.geojson, .csv: decimal long/lat)]]"

File: boundary_from_file.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+def boundary_from_file(file):
+    
+    #extension extraction out of file name
+    extension=file.split('.')[-1]
+    
+    #data loading
+    if extension=='geojson':
+        with open(file,encoding='utf8') as f:
+            data = json.load(f)
+            coordinates=data['features'][0]['geometry']['coordinates']
```
Comment (wehrad) [inline boundary_from_file.py:25]: add under line 25:
if len(np.shape(coordinates))!=2:
                coordinates=data['features'][0]['geometry']['coordinates'][0]
                if len(np.shape(coordinates))!=2:
                    coordinates=data['features'][0]['geometry']['coordinates'][0][0]

File: boundary_from_file.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    if extension=='csv':
+        coordinates=pd.read_csv(file)
+    else:
+        print('ERROR: %s extension not implemented' %extension)
+        return
+    
+    #boundary initialisation
+    boundary=''
+     
+    #coordinates reassignment if geojson file has a different organisation
```
Comment (wehrad) [inline boundary_from_file.py:35]: delete the try/except (line 35 to 39). It's now implemented under line 25

**Review by wehrad — COMMENTED** [review]
Hi Ken,

The documentation has been modified to precise the csv format. The structure of geojson files is always the same but the coordinates shape can change. Added to the normal shape, only one other case was considered (line 35 to 39). Now, I implemented new conditions to set coordinates variable correctly in all cases (under line 25).

**Comment by mankoff** [comment]
I don't see any mention anywhere about (lon,lat) or (lat,lon) for CSV file. What documentation? Is a file missing from the commit?

**Pull request merged**